### DISCUSSION TITLE
Send custom shortcuts to guest

### DISCUSSION THREAD

**Discussion by Andrew15-5** [quickemu-project/quickemu#1477] [Quickemu] [answered]
**I confirm this feature has not been previously requested**
- [x] I have searched the issues and this feature has not previously been requested

**Is your feature request related to a problem? Please describe.**
I want to switch to a tty1 (and others) inside guest (Linux), but instead I can only switch to it in host (Linux).

**Describe the solution you'd like**
Probably something similar to https://askubuntu.com/q/54814. I don't really care.

**Describe alternatives you've considered**
I don't see any alternatives.

**Additional context**
None.

**Comment by lj3954** [comment]
Quickemu's monitor-cmd flag is probably what you're looking for. Or, you can manually connect to the QEMU monitor, that quickemu lists when you launch a VM, with a tool of your choosing.

**Reply by Andrew15-5** [reply]
If I have a running VM, how would I use the monitor-cmd flag to send a shortcut to the specific VM?

**Reply by lj3954** [reply]
`quickemu --vm ${CONF_FILE} --monitor-cmd ${CMD}`

**Reply by Andrew15-5** [reply, marked as answer]
<details><summary>It can't be that easy.</summary>

![image](https://github.com/user-attachments/assets/e728c4c0-b825-447d-88aa-6a27779c2a8b)

</details>

Wow, I thought it is more complicated that that. I also found out that I have to make the last few arguments a single one:

```just
conf := "nixos.conf"
default:
  quickemu --vm '{{conf}}' --display spice

cmd *args:
  quickemu --vm '{{conf}}' --monitor-cmd '{{args}}'

alias sh := shortcut
shortcut *args:
  quickemu --vm '{{conf}}' --monitor-cmd 'sendkey {{args}}'

tty n="7":
  quickemu --vm '{{conf}}' --monitor-cmd 'sendkey ctrl-alt-f{{n}}'
```

I also found the example for this exact use case in https://www.qemu.org/docs/master/system/monitor.html.

Now I can easily switch between ttys with a simple `j tty 1`.

Thank you for your support.
